src/utils/getMessage.js

```diff
@@ -0,0 +1,24 @@
+import languages from '../i18n';
+import getAttr from './getAttr';
+import config from '../config';
+
+/* eslint no-param-reassign: ["error", { "props": false }]*/
+
+/**
+ * Uses getAttr to obtain translations for UI message strings outside react-intl.
+ * @param  {string} string - The identifier for the string to be localized
+ * @param  {string} lang - Language code for requested language
+ */
+
+const getMessage = (string, lang = config.activeLanguage) => {
+  // construct the translations object that only contains existing translations for given string
```

Comment:
Okay, um, I don't see a reason for using `reduce` here. Remember that `getMessage()` can easily be a fast-path function (i.e. called often), so it should be fast.

Maybe something like this (with old-school `for` loops so we get more performance and the ability to exit early).

```javascript
const getMessage = (string, lang = config.activeLanguage) => {
  // The usual path – the language exists and the string is translated in it.
  if(languages[lang] && languages[lang][string]) return languages[lang][string];
  // Otherwise, return the text in the first available language...
  for(let i = 0; i < config.languages.length; i++) {
    const map = languages[config.languages[i]];
    if(map && map[string]) return map[string];
  }
  // Or if that fails, just return the original.
  return string;
}